app/main/activity/crowdsourced-survey-table.directive.js

```diff
@@ -0,0 +1,97 @@
+module.exports = CrowdsourcedSurveyTable;
+
+function CrowdsourcedSurveyTable() {
+    return {
+        restrict: 'E',
+        scope: {
+            filters: '='
+        },
+        controller: CrowdsourcedSurveyTableController,
+        template: require('./crowdsourced-survey-table.html')
+    };
+}
+
+CrowdsourcedSurveyTableController.$inject = ['$scope', '$translate', 'FormEndpoint', 'FormStatsEndpoint', 'FormContactEndpoint', 'PostEndpoint', '_', 'PostFilters', 'Session'];
+function CrowdsourcedSurveyTableController($scope, $translate, FormEndpoint, FormStatsEndpoint, FormContactEndpoint, PostEndpoint, _, PostFilters, Session) {
+    $scope.crowdsourcedSurveyStatsByForm = [];
+    $scope.totalAllReduced = 0;
+    $scope.totalWebReduced = 0;
+    $scope.totalEmailReduced = 0;
+    $scope.totalSmsReduced = 0;
+    $scope.totalTwitterReduced = 0;
+    $scope.noSurveys = false;
+    $scope.unstructuredStats = {};
+
+    // whenever the filters changes, update the current list of posts
+    // this also initiates the first call to get form stats because
+    // it is called when the filters load the first time
+    $scope.$watch('filters', function () {
+        clearStats();
+        getFormStats();
+    }, true);
+    PostFilters.setMode('activity');
+
+    function clearStats() {
+        $scope.crowdsourcedSurveyStatsByForm = [];
+        $scope.totalAllReduced = 0;
+        $scope.totalWebReduced = 0;
+        $scope.totalEmailReduced = 0;
+        $scope.totalSmsReduced = 0;
+        $scope.totalTwitterReduced = 0;
+        $scope.unstructuredStats = {};
+    }
+
+    function getFormStats() {
+        FormEndpoint.query().$promise.then((forms) => {
+            if (crowdsourcedSurveysExist) {
+                $scope.noSurveys = false;
+                forms.forEach((form) => {
+                    if (!form.targeted_survey) {
+                        let query = PostFilters.getQueryParams($scope.filters) || {};
+                        var postQuery = _.extend({}, query, {
```

Comment:
I think postQuery should be using `let` here and take advantage of block level scope, specially because this requests won't necessarily all run before the unstructuredStats section below and can cause issues

Reply:
as a side note:  maybe we want to take advantage of promise.all here?

Reply:
I think the promise.all would be tough because we're not getting back the form name with the stats. So it would be a pain to add the form name to each result after we get them all back.

Reply:
mmmm.  OK, I think (think!) this would work anyway as long as the `var` is changed `let` for some block scope sanity 👍